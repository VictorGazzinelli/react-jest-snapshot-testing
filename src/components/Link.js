import React from 'react';

const Link = (props => (
    <a id={props.id} href={props.url}> {props.text} </a>
));

export default Link;

