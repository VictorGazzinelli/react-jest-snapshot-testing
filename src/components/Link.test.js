import React from 'react';
import renderer from 'react-test-renderer';
import Link from './Link';

it(`renders correctly with given props`, () => {
  const linkJsonSnapshot = 
  renderer.create(<Link id={'myNewLink'} url={'https://reactjs.org/docs/getting-started.html'} text={'Get Started'} />)
    .toJSON();
  expect(linkJsonSnapshot).toMatchSnapshot();
});