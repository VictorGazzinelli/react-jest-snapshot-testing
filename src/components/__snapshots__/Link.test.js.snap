// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with given props 1`] = `
<a
  href="https://reactjs.org/docs/getting-started.html"
  id="myNewLink"
>
   
  Get Started
   
</a>
`;
